### PR TITLE
[WIP] Learning rate schedule

### PR DESCRIPTION
**Pull request by ruimashita**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature / Optimization (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: blueoil/blueoil_init.py [hunk 1]
```diff
@@ -101,6 +102,14 @@
 # ]
 
 
+learning_rate_schedule_map = OrderedDict({
+    "constant": "'constant' -> constant learning rate.",
+    "2-step-decay": "'2-step-decay' -> learning rate reduce to 1/10 on epochs/2 and epochs-1.",
```
Comment (lm-konda) [inline blueoil/blueoil_init.py]: I think `epochs` is ambiguous.

File: tests/config/make_yml_config.py [hunk 2]
```diff
@@ -113,21 +113,23 @@
...
-    '  learning_rate_setting: tune1\n',
-    '  learning_rate_setting: tune1\n',
-    '  learning_rate_setting: tune1\n',
-    '  learning_rate_setting: tune1\n',
-    '  learning_rate_setting: tune1\n',
-    '  learning_rate_setting: tune1\n',
+trainer_lr_schedule_comment = """\
+  # supported 'learning_rate_schedule' is 'constant', '2-step-decay', '3-step-decay', '3-step-decay-with-warmup'.
+  #   'constant' -> constant learning rate.
+  #   '2-step-decay' -> learning rate reduce to 1/10 on epochs/2 and epochs-1.
```
Comment (lm-konda) [inline tests/config/make_yml_config.py]: epochs is ambiguous too.

File: tests/config/make_yml_config.py [hunk 3]
```diff
@@ -113,21 +113,23 @@
...
+  #   '3-step-decay' -> learning rate reduce to 1/10 on epochs/3 and epochs*2/3 and epochs-1
```
Comment (lm-konda) [inline tests/config/make_yml_config.py]: I prefer "decrease" to "reduce".

File: blueoil/blueoil_init.py (hunk 1 above)
Comment (lm-jira) [reply blueoil/blueoil_init.py]: I think `learning rate reduce to 1/10` means lr is changed to 0.1.
It may be better to change to `learning rate is reduce by 1/10`.

**Comment by ruimashita** [comment]
@lm-jira @lm-konda 
Please review, I changed the text more readable in https://github.com/blue-oil/blueoil/pull/82/commits/ea8f63aa5b530401a43f9002c4ffa0fc7ee9a4d4

File: blueoil/blueoil_init.py (hunk 1 above)
Comment (lm-konda) [reply blueoil/blueoil_init.py]: Maybe `number of epochs` is popular.
How about this?

Comment (ruimashita) [reply blueoil/blueoil_init.py]: Sure!!

Comment (ruimashita) [reply blueoil/blueoil_init.py]: @lm-konda
but I already changed messages using `{epochs}`
https://github.com/blue-oil/blueoil/blob/ea8f63aa5b530401a43f9002c4ffa0fc7ee9a4d4/blueoil/blueoil_init.py#L308

‘number of epochs’ is better than `{epochs}` ?

Comment (lm-konda) [reply blueoil/blueoil_init.py]: I think there is no change from the state that the meaning is ambiguous in essence.

Comment (ruimashita) [reply blueoil/blueoil_init.py]: @lm-konda 
Is the below message ambiguous for you (and user)?
https://github.com/blue-oil/blueoil/blob/ea8f63aa5b530401a43f9002c4ffa0fc7ee9a4d4/blueoil/blueoil_init.py#L308

Comment (lm-konda) [reply blueoil/blueoil_init.py]: I was worried about how to change, but now I feel good even as it is.

File: blueoil/blueoil_init.py [hunk 4]
```diff
@@ -101,6 +102,14 @@
 # ]
 
 
+learning_rate_schedule_map = OrderedDict([
+    ("constant", "'constant' -> constant learning rate."),
+    ("2-step-decay", "'2-step-decay' -> learning rate decrease by 1/10 on {epochs}/2 and {epochs}-1."),
+    ("3-step-decay", "'3-step-decay' -> learning rate decrease by 1/10 on {epochs}/3 and {epochs}*2/3 and {epochs}-1"),
+    ("3-step-decay-with-warmup", "'3-step-decay-with-warmup' -> warmup learning rate 1/1000 in first epoch, then train same as '3-step-decay'"),
```
Comment (lm-jira) [inline blueoil/blueoil_init.py]: I think `train the same way as '3-step-decay'` sounds more natural than `train same as '3-step-decay'`.

Comment (ruimashita) [reply blueoil/blueoil_init.py]: @lm-jira thanks

Comment (ruimashita) [reply blueoil/blueoil_init.py]: @lm-jira I changed message to `train the same way as '3-step-decay'` in 1b82b44
